src/handler/httpfuture.rs

```diff
@@ -0,0 +1,65 @@
+use futures::channel::mpsc;
+use pin_project_lite::pin_project;
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use crate::error::Result;
+use crate::handler::commandfuture::CommandFuture;
+use crate::handler::http::HttpRequest;
+use crate::handler::navigationfuture::NavigationFuture;
+use crate::handler::target::TargetMessage;
+use chromiumoxide_types::Command;
+
+type ArcRequest = Option<Arc<HttpRequest>>;
+
+pin_project! {
+    pub struct HttpFuture<T: Command> {
+        #[pin]
+        command: CommandFuture<T>,
+        #[pin]
+        navigation: NavigationFuture,
+
+        command_complete: bool,
```

Comment:
you could use `Fuse<CommandFuture>` instead of the separate bool here

Reply:
good idea 👍